### ISSUE TITLE
nodata not set in output DEM

### ISSUE THREAD

**Issue by dshean** [uw-cryo/opentopo_dem#6]
`download_global_DEM.py -demtype COP30_E -raster_fn CO_WestCentral_2019_overlap_bbox_EarthDEM_dem_shpclip.tif -out_proj EPSG:32613`

Extent of raster is: 
UTM 13N: `347466.2026631 4270393.08631403 378670.2026631 4315705.08631403`
4326: `38.568909623145416 -106.76090090473916 38.981935661144625 -106.39282064394672`

Output looks, good, but `gdalinfo COP30_E.tif` indicates nodata is not set, and subsequent dem_align run has spike of 0 diff values as a result.  Should be simple fix, can investigate after AGU.

**Comment by ShashankBice** [comment]
Thanks, maybe run after git pull, I put [a fix](https://github.com/uw-cryo/fetch_dem/commit/ae1a03041677b02157b5564b99496f5495c6031a) in November, and all my tests have shown no-data is properly assigned in the output :)

**Comment by dshean** [comment]
OK, you're right, I was not using the latest.  Did not test, but seems like this should work.

I seem to recall an earlier conversation about nodata handling for fetch_dem, but I can't find it now.  https://github.com/uw-cryo/fetch_dem/commit/ae1a03041677b02157b5564b99496f5495c6031a was a "quick fix", but in the future we should try to preserve the nodata value of the original DEM source, and avoid reliance on bash calls to Python utilities like gdal_edit.py, when we can open the GDAL dataset and directly set the metadata in Python. 

 But plenty of other higher priorities right now :) Can revisit later, perhaps as the 3D CRS handling evolves.

**Comment by ShashankBice** [comment]
Yes, you are right, I assign the nodata value to the original DEM. The original DEM in EPSG:4326 has no nodata, so they do not have any nodata value assigned there. I assign it in the EPSG:4326 downloaded file, and it is carried forward to the final reprojected DEM. 
Re subprocess calls, I am going to replace those instances with gdal as I make the package pip installable next week.

Cheers!

**Comment by dshean** [comment]
OK, I think this is pretty low priority, as current tools are working for our needs.  Cleaning up for release is a good goal, but will inevitably to take more time than expected.
